src/routes/token.rs

```diff
@@ -46,6 +47,22 @@ async fn post_token_handler(
         .await
         .ok_or(Error::InvalidAuthorizationCode)?;
 
+    // XNOR because client_secret and client_id must have the same Option value.
+    if body.client_secret.is_some() != body.client_id.is_some() {
```

Comment:
I think it would be more readable to use a match statement already binding it to the expected value. 

Something like

```rust
match (body.client_secret, body.client_id) {
    (Some(_), Some(client_id)) if client_id == response_metadata.client_id => Ok,
    (Some(_), Some(_)) => ...
    (Some(_), None) => ...
    (None, Some(_)) => ...
```

This is a perfect use case for this construct.

Reply:
[b1f3fe7](https://github.com/KILTprotocol/opendid/pull/53/commits/b1f3fe79a199cbaa274c603d0978d8c19e1e69b8)

Actually I wrote it first using match and then changed it to how it was because I thought it was nicer, but I don't insist.